### ISSUE TITLE
KOTOR: Error: unable to detect the game in Haiku_Os

### ISSUE THREAD

**Issue by johna23-lab** [xoreos/xoreos#594]
I compiled the code in Linux (Mint 19.3) and the engine and game went perfect, then i decided to compile the code under Haiku_Os, and the compilation went perfect too, but is unable to detect the game

Any idea?

Thanks!

**Comment by Peppersawce** [comment]
Using NWN's game data, I had to extract it twice but tampered with the game folder the first time.
When using the gog installer via innoextract the actual game files are those in the "game" subfolder.

About xoreos on Haiku:
Built from the latest source without patches but it needs to be fed the path to the SDL2 headers during configuration.
Auto-generates its own settings file and puts it in the right place when it starts up.
I suppose it's meant to produce logs but I'm not seeing them anywhere atm.
![Image](https://github.com/user-attachments/assets/fb7a55f7-2233-43da-81c8-d610bfe48970)
Also, it thinks it's on Windows for some reason.

But I think the original issue is not valid, paths can be tricky sometimes.

**Comment by DrMcCoy** [comment]
> When using the gog installer via innoextract the actual game files are those in the "game" subfolder.

Yeah, I haven't really kept up with what the more modern rereleases do. The game detection is still based on the original layout of the game data, as created by the original installer on the CD.

> Built from the latest source without patches but it needs to be fed the path to the SDL2 headers during configuration.

Using our autotools- or our CMake-based build?

> I suppose it's meant to produce logs but I'm not seeing them anywhere atm.

They're on stdout/stderr, but also written to a file in a xoreos subfolder in XDG user data something path, which defaults to ~/.local/share/.

> Also, it thinks it's on Windows for some reason

No, it detected the Windows version of NWN (because it found the nwmain.exe, IIRC)

As for the original issue, oops, I guess I missed that one. But I don't really have any idea what could have gone wrong there, especially with not much info to go on.

**Comment by Peppersawce** [comment]
> Using our autotools- or our CMake-based build?
I used CMake, the script to find SDL2 fails because in Haiku they're in "headers/SDL2" in the system devel dir (have to recheck the path)
> No, it detected the Windows version of NWN (because it found the nwmain.exe, IIRC)
Ah, that makes sense. Also, that explains why it didn't work the first time, I just assumed the original executables weren't needed.

**Comment by DrMcCoy** [comment]
>  I just assumed the original executables weren't needed.

For some of the games, they're even vital, because there's data in the EXE resource structure. IIRC the mouse cursors for The Witcher, for example

**Comment by Peppersawce** [comment]
That's so weird, one cannot help but wonder why they'd do that.

About the original issue, I don't have Kotor so I can't help much more, but I'm probably going to make a recipe for the Haiku repos so more people can toy with xoreos on the system.

**Comment by Begasus** [comment]
This should fix detecting SDL2 on Haiku:

```diff
diff --git a/cmake/FindSDL2.cmake b/cmake/FindSDL2.cmake
index b6658fe..57d9b30 100644
--- a/cmake/FindSDL2.cmake
+++ b/cmake/FindSDL2.cmake
@@ -29,7 +29,8 @@ if(WIN32)
   find_library(SDL2_LIBRARY NAMES SDL2 PATHS $ENV{PROGRAMFILES}/SDL2/lib DOC "The SDL2 library")
 
 else(WIN32)
-  find_path(SDL2_INCLUDE_DIR SDL.h HINTS /usr/include/SDL2 /usr/local/include/SDL2 /opt/local/include/SDL2 /opt/homebrew/include/SDL2 DOC "The directory where SDL.h resides")
+  find_path(SDL2_INCLUDE_DIR SDL.h HINTS /usr/include/SDL2 /usr/local/include/SDL2 /opt/local/include/SDL2 /opt/homebrew/include/SDL2 
+    /boot/system/develop/headers/SDL2 /boot/system/develop/headers/x86/SDL2  DOC "The directory where SDL.h resides")
   find_library(SDL2_LIBRARY NAMES SDL2 DOC "The SDL2 library")
 
 endif(WIN32)
```
